### PR TITLE
Implement Trending Videos Feature with Category Selection and Its Tests

### PR DESCRIPTION
**Pull request by cyn900**
This merge introduces the 'Trending' feature, allowing users to see the top three trending YouTube videos based on selected categories (general, movies, music, sports, or gaming). Key highlights:

-**Trending Category Selection**: Users can now select their preferred category from the TrendingCategorySelect view, accessible from the home screen by clicking the Trending button.
-**Top 3 Trending Videos Display**: The application fetches and displays the top 3 trending videos as determined by YouTube's algorithm for the chosen category.
-**Factory Design Pattern**: Implemented a factory design pattern to create a TrendingCategorySelectView.
-**Comprehensive Testing**: Comprehensive Junit and Mockito test suites have been added to ensure reliability and maintainability:

- 100% test coverage for the use case interactor.
- Over 70% test coverage for other related components.

-**Enhanced User Experience**: The new feature is seamlessly integrated into the existing user interface, providing an intuitive and engaging experience for users seeking to discover trending content.

This feature enriches our application by offering dynamic, user-driven content discovery and aligns with our goal of providing an immersive and up-to-date media experience.

### CODE REVIEW

**Comment by jonathanjschen913** [comment]
Remember to use camel case naming instead of the Python underscore convention. Look into your data access interface and its usage files.

**Comment by AzkyT** [comment]
All seems good, although I do think some methods are missing some documentation. 

I also noticed in the signup page when you first open the application, it isn't very intuitive that the cancel button should bring you to the login screen, maybe that's something we should change in the future.

File: src/views/TrendingCategorySelectView.java
[the remark this answers is not in the document]
Comment (cyn900) [reply src/views/TrendingCategorySelectView.java:1]: fixed

**Comment by jonathanjschen913** [comment]
👍

**Pull request merged**